client/my-sites/stats/annual-site-stats/index.js

```diff
@@ -0,0 +1,113 @@
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	isRequestingSiteStatsForQuery,
+	getSiteStatsNormalizedData,
+} from 'state/stats/lists/selectors';
+import StatsModulePlaceholder from 'my-sites/stats/stats-module/placeholder';
+import ErrorPanel from 'my-sites/stats/stats-error';
+
+class AnnualSiteStats extends Component {
+	static propTypes = {
+		requesting: PropTypes.bool,
+		siteId: PropTypes.number,
+		years: PropTypes.array,
+		translate: PropTypes.func,
+		moment: PropTypes.func,
+	};
+
+	renderContent( data ) {
+		const { translate } = this.props;
+		return (
+			<div className="annual-site-stats__content">
+				<div className="annual-site-stats__stat">
+					<div className="annual-site-stats__stat-title">{ translate( 'year' ) }</div>
+					<div className="annual-site-stats__stat-figure is-large">{ data.year }</div>
+				</div>
+				<div className="annual-site-stats__stat">
+					<div className="annual-site-stats__stat-title">{ translate( 'total posts' ) }</div>
+					<div className="annual-site-stats__stat-figure is-large">{ data.total_posts }</div>
+				</div>
+				<div className="annual-site-stats__stat">
+					<div className="annual-site-stats__stat-title">{ translate( 'total comments' ) }</div>
+					<div className="annual-site-stats__stat-figure">{ data.total_comments }</div>
+				</div>
+				<div className="annual-site-stats__stat">
+					<div className="annual-site-stats__stat-title">
+						{ translate( 'avg comments per post' ) }
+					</div>
+					<div className="annual-site-stats__stat-figure">{ data.avg_comments }</div>
+				</div>
+				<div className="annual-site-stats__stat">
+					<div className="annual-site-stats__stat-title">{ translate( 'total likes' ) }</div>
+					<div className="annual-site-stats__stat-figure">{ data.total_likes }</div>
+				</div>
+				<div className="annual-site-stats__stat">
+					<div className="annual-site-stats__stat-title">{ translate( 'avg likes per post' ) }</div>
+					<div className="annual-site-stats__stat-figure">{ data.avg_likes }</div>
+				</div>
+				<div className="annual-site-stats__stat">
+					<div className="annual-site-stats__stat-title">{ translate( 'total words' ) }</div>
+					<div className="annual-site-stats__stat-figure">{ data.total_words }</div>
+				</div>
+				<div className="annual-site-stats__stat">
+					<div className="annual-site-stats__stat-title">{ translate( 'avg words per post' ) }</div>
+					<div className="annual-site-stats__stat-figure">{ data.avg_words }</div>
+				</div>
+			</div>
+		);
+	}
+
+	render() {
+		const { years, requesting, siteId, translate, moment, statType } = this.props;
+		const currentYear = moment().format( 'YYYY' );
+		const year = years && find( years, y => y.year === currentYear );
+		const isLoading = requesting && ! years;
+		/* eslint-disable wpcalypso/jsx-classname-namespace */
+		return (
+			<div>
+				{ siteId && <QuerySiteStats siteId={ siteId } statType={ statType } /> }
```

Comment:
If we are only going to use this component in the context of the Insights page, we could reasonably rely upon the "Post Trends" component already making the query to the `statsInsights` endpoint. On this branch that endpoint is called 2x now on the page load with the same query params.

If we consider adding this component to the Years period page, then we would need to add the query component to ensure we have data to work with.